driver/pace/driver/run.py

```diff
@@ -279,11 +390,42 @@ def total_time(self) -> timedelta:
             days=self.days, hours=self.hours, minutes=self.minutes, seconds=self.seconds
         )
 
+    @staticmethod
+    def _set_kwargs_from_serialbox(kwargs: Dict[str, Any], config: SerialboxConfig):
+        kwargs["nx_tile"] = config.namelist.npx - 1
+        kwargs["nz"] = config.namelist.npz
+        kwargs["layout"] = tuple(config.namelist.layout)
+        kwargs["dt_atmos"] = float(config.namelist.dt_atmos)
+        kwargs["dycore_config"] = fv3core.DynamicalCoreConfig.from_f90nml(
+            config.f90_namelist
+        )
+        kwargs["physics_config"] = fv3gfs.physics.PhysicsConfig.from_f90nml(
+            config.f90_namelist
+        )
+        kwargs["days"] = config.namelist.days
+        kwargs["hours"] = config.namelist.hours
+        kwargs["minutes"] = config.namelist.minutes
+        kwargs["seconds"] = config.namelist.seconds
+        return kwargs
+
     @classmethod
     def from_dict(cls, kwargs: Dict[str, Any]) -> "DriverConfig":
-        kwargs["layout"] = tuple(kwargs["layout"])
         initialization_type = kwargs["initialization_type"]
-        if initialization_type == "baroclinic":
+        if initialization_type == "serialbox":
+            config = kwargs.get("initialization_config", {})
+            config.update(kwargs.get("stencil_config", {}))
```

Comment:
Are these lines necessary? My understanding was that dacite does this automatically, but I could be wrong.
```suggestion
```

Reply:
It is necessary because we're making a SerialboxConfig first before making DriverConfig class, and we need backend from stencil_config.